cpp/arcticdb/column_store/memory_segment_impl.hpp

```diff
@@ -262,6 +263,37 @@ class SegmentInMemoryImpl {
             return row_id_ == other.row_id_ && parent_ == other.parent_;
         }
 
+        size_t hash() const{
```

Comment:
All of this needs to go elsewhere. It would be nice to move the segment and row iterators out of the segment file as well, they are somewhat useful in that we can do things like std::shuffle a segment in tests, but they aren't the way that people should be interacting with segment data in general, they should use ColumnData and the iterators that @alexowens90 has created